sunpy/timeseries/sources/noaa.py

```diff
@@ -18,7 +18,168 @@
+            noaa.peek()
+        Parameters
+        ----------
+        type : {'GOES-Long&Short', 'GOES-Long', 'GOES-Short'}, optional
+            The type of plot required. Defaults to "GOES-Long&Short".
+        **kwargs : `dict`
+            Additional plot keyword arguments that are handed to `axes.plot` functions
+        """
+        # Check we have a timeseries valid for plotting
+        self._validate_data_for_plotting()
+
+        figure = plt.figure()
+        axes = plt.gca()
+        # dataframe = self.to_dataframe()
+
+        if type == 'GOES-Long_and_Short':
+            dataframe_long = self._split_to_dataframe(type='GOES-Long')
+            axes = dataframe_long['flux'].plot(marker='', color='red',
+                                               linewidth=1, label="GOES-Long", **plot_args)
+            dataframe_sort = self._split_to_dataframe(type='GOES-Short')
+            dataframe_sort['flux'].plot(marker='', color='blue',
+                                        linewidth=1, label="GOES-Short", **plot_args)
+            axes.set_ylabel(r'Flux [$Watts \cdot m^{-2}$]')
+        elif type == 'GOES-Long':
+            dataframe_long = self._split_to_dataframe(type='GOES-Long')
+            axes = dataframe_long['flux'].plot(marker='', color='red',
+                                               linewidth=1, label="GOES-Long", **plot_args)
+            axes.set_ylabel(r'Flux [$Watts \cdot m^{-2}$]')
+        elif type == 'GOES-Short':
+            dataframe_sort = self._split_to_dataframe(type='GOES-Short')
+            axes = dataframe_sort['flux'].plot(marker='', color='blue',
+                                               linewidth=1, label="GOES-Short", **plot_args)
+            axes.set_ylabel(r'Flux [$Watts \cdot m^{-2}$]')
+        else:
+            raise ValueError(f'Got unknown plot type "{type}"')
+
+        axes.set_title('NOAA - GOES Soft X-Ray Flux (1-minute average)')
+        axes.set_yscale('log')
+        axes.set_ylim([1e-9, 1e-2])
+        axes.grid(True, which='minor', linewidth=0.5)
+        axes.grid(True, which='major', linewidth=0.5)
+        ymin, ymax = axes.get_ylim()
+
+        # Add a color to the classes limits
+        ygrid = axes.get_ygridlines()
+        ygrid[2].set_color('blue')
+        ygrid[2].set_linewidth(1)
+        ygrid[3].set_color('green')
+        ygrid[3].set_linewidth(1)
+        ygrid[4].set_color('yellow')
+        ygrid[4].set_linewidth(1)
+        ygrid[5].set_color('orange')
+        ygrid[5].set_linewidth(1)
+        ygrid[6].set_color('red')
+        ygrid[6].set_linewidth(1)
+
+        # Add a label at the classes lines
+        ax2 = axes.twinx()
+        ax2.set_yscale("log")
+        ax2.set_ylim(ymin, ymax)
+        ax2.set_yticks((1e-9, 1e-8, 1e-7, 1e-6, 1e-5, 1e-4, 1e-3, 1e-2))
+        ax2.set_yticklabels((' ', 'A', 'B', 'C', 'M', 'X', ' '))
+        axes.get_shared_x_axes().join(axes, ax2)
+
+        return figure
```

Comment:
Can you show us what the figure looks like?

Reply:
After a few modifications that I haven't pushed yet (e.g. add legend +  labels in the center of the flare classes)
the figure looks like:
![GOES_SXR_ExampleV1](https://user-images.githubusercontent.com/10522585/87545359-1659a300-c6a8-11ea-8653-027db087788a.png)